webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantDialog.tsx

```diff
@@ -25,6 +26,11 @@ function CreateVariantDialog(props: Props) {
   const [sourceList, setSourceList] = useState<GenericInfo[]>([]);
   const defaultValues = { name: "", sourceId: parentId };
 
+  const existingVariants = useMemo(
```

Comment:
`variantNames` is simpler but less specific. 

It's specifically used for validation to ensure the name entered for a new variant is unique.
Given the context, `existingVariants` accurately describes the content of the variable: it is an array of names representing variants that already exist. This is important for understanding the purpose of the variable, especially in the context of form validation where it's used to prevent the creation of a variant with a duplicate name. (same for the others similar comments)